### PR TITLE
Update go.mod version for telemetry build to align with sonic-slave-bullseye

### PR DESCRIPTION
**Pull request by FengPan-Frank**
#### Why I did it
Go before 1.15.13 and 1.16.x before 1.16.5 has functions for DNS lookups that do not validate replies from DNS servers, and thus a return value may contain an unsafe injection (e.g., XSS) that does not conform to the RFC1035 format. Now 1.15.15 is used in sonic-slave-bullseye thus it's safe, this change is to update go.mod for align dependency with build version meanwhile.

#### How I did it
Update go.mod align with golang compiler used by sonic-slave-bullseye docker.

#### How to verify it
unit test to do sanity check.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### Description for the changelog

Add golang version check in gnmi build process to make sure no security concern later, for CVE-2021-33195

### CODE REVIEW

**Review by ganglyu — DISMISSED** [review]

**Review by zbud-msft — APPROVED** [review]

**Comment by ganglyu** [comment]
Please update PR checker to make sure binary is build with go 1.18

File: Makefile [hunk 1]
```diff
@@ -61,6 +69,19 @@ go-deps-clean:
 	$(RM) -r vendor
 
 sonic-gnmi: $(GO_DEPS)
+	$(GO) version
```
[the remark this answers is not in the document]
Comment (FengPan-Frank) [reply Makefile:72]: Debugging purpose, will remove it.

**Comment by prsunny** [comment]
@theasianpianist for viz

**Review by ganglyu — APPROVED** [review]

**Pull request merged**